webviews/common/context.tsx

```diff
@@ -176,6 +176,17 @@ export class PRContext {
 
 	public openDiff = (comment: IComment) => this.postMessage({ command: 'pr.open-diff', args: { comment } });
 
+	public toggleResolveComment = (reviewThread: ReviewEvent['reviewThread'], thread: IComment[], newResolved: boolean) => {
+		if (!reviewThread) {
+			return;
+		}
+		reviewThread.isResolved = newResolved;
+		reviewThread.canUnresolve = newResolved;
+		reviewThread.canResolve = !newResolved;
+		this.updatePR({});
```

Comment:
The usual patter for `updatePR` is to pass the properteis of the PR that changed into `updatePR`. Does just calling it without passing in the updated `{ events }` work?

Reply:
I guess you're modifying the original object, so that's why it would work. I find this to be a little tricky to understand and not self-documenting (I have to make sure I understand that `reviewThread` is the same object that is already in `this.pr`, there are no clues about what is happening). Instead, is it possible to use the usual `const response = await this.postMessage` then pass some property of the `response` into `updatePR`?

Reply:
It seemed to work but I just noticed it actually doe not. I'm trying to figure out how to only update the state of one array item

Reply:
Would it make sense to call `PullRequestModel`'s `getTimelineEvents` method after doing the resolve in `PullRequestOverViewPanel`? That would also have the benefit of updating state that the overview uses so that we're up-to-date in both the extension and the webview as well as giving you the correct item to include in the message response.